### DISCUSSION TITLE
using qnap to install, but getting an openvpn error

### DISCUSSION THREAD

**Discussion by kyrusdemnati** [haugene/docker-transmission-openvpn#2176] [General]
i am using the contrainer station,

Starting container with revision: 20877f1b168b6ff27fc58aeef40756572c562d47
Creating TUN device /dev/net/tun
mknod: /dev/net/tun: File exists
ERROR: Cannot determine where to find your OpenVPN config. Both OPENVPN_CONFIG_URL and OPENVPN_PROVIDER is unset.
You have to either provide a URL to the config you want to use, or set a configured provider that will download one for you.
Exiting...

please help, as i would love to use tranmission with a secure vpn via a docker image

**Comment by pkishino** [comment]
a few comments:
- qnap is notoriously bad for this as it actively tries to kill the openvpn process from what I heard
- please post your complete config you are using
- please read the documentation for this container on how to set it up

**Reply by kyrusdemnati** [reply]
i am getting this error, i am installing via qnap.

Modification: Change tls-crypt keyfile path
Modification: Set output verbosity to 3
Modification: Remap SIGUSR1 signal to SIGTERM, avoid OpenVPN restart loop
Setting OpenVPN credentials...
adding route to local network 10.0.0.0/24 via 10.0.3.1 dev eth0
Sat Feb  5 10:15:36 2022 disabling NCP mode (--ncp-disable) because not in P2MP client or server mode
Options error: You must define TUN/TAP device (--dev)
Use --help for more information.
Starting container with revision: 20877f1b168b6ff27fc58aeef40756572c562d47
Creating TUN device /dev/net/tun
Using OpenVPN provider: AIRVPN
Running with VPN_CONFIG_SOURCE auto
No bundled config script found for AIRVPN. Defaulting to external config
Downloading configs from https://github.com/haugene/vpn-configs-contrib/archive/main.zip into /tmp/tmp.ivV3JZzpKR
Extracting configs to /tmp/tmp.jxg1zg8cJ4
ERROR: Could not find any configs for provider AIRVPN in downloaded configs
Cleanup: deleting /tmp/tmp.ivV3JZzpKR and /tmp/tmp.jxg1zg8cJ4
No VPN configuration provided. Using default.
Modifying /etc/openvpn/airvpn/default.ovpn for best behaviour in this container
Modification: Point auth-user-pass option to the username/password file
Modification: Change ca certificate path
Modification: Change ping options
Modification: Update/set resolv-retry to 15 seconds
Modification: Change tls-crypt keyfile path
Modification: Set output verbosity to 3
Modification: Remap SIGUSR1 signal to SIGTERM, avoid OpenVPN restart loop
Setting OpenVPN credentials...
adding route to local network 10.0.0.0/24 via 10.0.3.1 dev eth0
Sat Feb  5 10:16:11 2022 disabling NCP mode (--ncp-disable) because not in P2MP client or server mode
Options error: You must define TUN/TAP device (--dev)
Use --help for more information.
Starting container with revision: 20877f1b168b6ff27fc58aeef40756572c562d47
Creating TUN device /dev/net/tun
Using OpenVPN provider: AIRVPN
Running with VPN_CONFIG_SOURCE auto
No bundled config script found for AIRVPN. Defaulting to external config
Downloading configs from https://github.com/haugene/vpn-configs-contrib/archive/main.zip into /tmp/tmp.4hykIluh0A
Extracting configs to /tmp/tmp.cPHYtbCWFC
ERROR: Could not find any configs for provider AIRVPN in downloaded configs
Cleanup: deleting /tmp/tmp.4hykIluh0A and /tmp/tmp.cPHYtbCWFC
No VPN configuration provided. Using default.
Modifying /etc/openvpn/airvpn/default.ovpn for best behaviour in this container
Modification: Point auth-user-pass option to the username/password file
Modification: Change ca certificate path
Modification: Change ping options
Modification: Update/set resolv-retry to 15 seconds
Modification: Change tls-crypt keyfile path
Modification: Set output verbosity to 3
Modification: Remap SIGUSR1 signal to SIGTERM, avoid OpenVPN restart loop
Setting OpenVPN credentials...
adding route to local network 10.0.0.0/24 via 10.0.3.1 dev eth0
Sat Feb  5 10:17:11 2022 disabling NCP mode (--ncp-disable) because not in P2MP client or server mode
Options error: You must define TUN/TAP device (--dev)
Use --help for more information.
Starting container with revision: 20877f1b168b6ff27fc58aeef40756572c562d47
Creating TUN device /dev/net/tun
Using OpenVPN provider: AIRVPN
Running with VPN_CONFIG_SOURCE auto
No bundled config script found for AIRVPN. Defaulting to external config
Downloading configs from https://github.com/haugene/vpn-configs-contrib/archive/main.zip into /tmp/tmp.MwTxjieE4M
Extracting configs to /tmp/tmp.lfDd2R0SNZ
ERROR: Could not find any configs for provider AIRVPN in downloaded configs
Cleanup: deleting /tmp/tmp.MwTxjieE4M and /tmp/tmp.lfDd2R0SNZ
No VPN configuration provided. Using default.
Modifying /etc/openvpn/airvpn/default.ovpn for best behaviour in this container
Modification: Point auth-user-pass option to the username/password file
Modification: Change ca certificate path
Modification: Change ping options
Modification: Update/set resolv-retry to 15 seconds
Modification: Change tls-crypt keyfile path
Modification: Set output verbosity to 3
Modification: Remap SIGUSR1 signal to SIGTERM, avoid OpenVPN restart loop
Setting OpenVPN credentials...
adding route to local network 10.0.0.0/24 via 10.0.3.1 dev eth0
Sat Feb  5 10:18:20 2022 disabling NCP mode (--ncp-disable) because not in P2MP client or server mode
Options error: You must define TUN/TAP device (--dev)
Use --help for more information.
Starting container with revision: 20877f1b168b6ff27fc58aeef40756572c562d47
Creating TUN device /dev/net/tun
Using OpenVPN provider: AIRVPN
Running with VPN_CONFIG_SOURCE auto
No bundled config script found for AIRVPN. Defaulting to external config
Downloading configs from https://github.com/haugene/vpn-configs-contrib/archive/main.zip into /tmp/tmp.3DlZzVgEqX
Extracting configs to /tmp/tmp.VpFb8OgH6G
ERROR: Could not find any configs for provider AIRVPN in downloaded configs
Cleanup: deleting /tmp/tmp.3DlZzVgEqX and /tmp/tmp.VpFb8OgH6G
No VPN configuration provided. Using default.
Modifying /etc/openvpn/airvpn/default.ovpn for best behaviour in this container
Modification: Point auth-user-pass option to the username/password file
Modification: Change ca certificate path
Modification: Change ping options
Modification: Update/set resolv-retry to 15 seconds
Modification: Change tls-crypt keyfile path
Modification: Set output verbosity to 3
Modification: Remap SIGUSR1 signal to SIGTERM, avoid OpenVPN restart loop
Setting OpenVPN credentials...
adding route to local network 10.0.0.0/24 via 10.0.3.1 dev eth0
Sat Feb  5 10:19:29 2022 disabling NCP mode (--ncp-disable) because not in P2MP client or server mode
Options error: You must define TUN/TAP device (--dev)
Use --help for more information.
Starting container with revision: 20877f1b168b6ff27fc58aeef40756572c562d47
                                                                          Creating TUN device /dev/net/tun
                                                                                                          Using OpenVPN provider: AIRVPN
   Running with VPN_CONFIG_SOURCE auto
                                      No bundled config script found for AIRVPN. Defaulting to external config
                                                                                                              Downloading configs from https://github.com/haugene/vpn-configs-contrib/archive/main.zip into /tmp/tmp.vYdyOhqW5o
                                                                                          Extracting configs to /tmp/tmp.CdCy0goQfs
                                                                                                                                   ERROR: Could not find any configs for provider AIRVPN in downloaded configs
                                                                         Cleanup: deleting /tmp/tmp.vYdyOhqW5o and /tmp/tmp.CdCy0goQfs
 No VPN configuration provided. Using default.
                                              Modifying /etc/openvpn/airvpn/default.ovpn for best behaviour in this container
                                                                                                                             Modification: Point auth-user-pass option to the username/password file
                                                               Modification: Change ca certificate path
                                                                                                       Modification: Change ping options
   Modification: Update/set resolv-retry to 15 seconds
                                                      Modification: Change tls-crypt keyfile path
                                                                                                 Modification: Set output verbosity to 3
   Modification: Remap SIGUSR1 signal to SIGTERM, avoid OpenVPN restart loop
                                                                            Setting OpenVPN credentials...
                                                                                                          adding route to local network 10.0.0.0/24 via 10.0.3.1 dev eth0
                                    Sat Feb  5 10:20:48 2022 disabling NCP mode (--ncp-disable) because not in P2MP client or server mode
    Options error: You must define TUN/TAP device (--dev)
                                                         Use --help for more information."

**Reply by pkishino** [reply]
just the log is not very helpful..please read my comment above and provide the config

**Reply by pkishino** [reply]
also..if you had checked the documentation you would have seen that airvpn is NOT on the provider list, so if you want to use it you have to setup a custom provider..ALSO detailed in the documentation.

**Comment by kyrusdemnati** [comment]
hi this is the log i have been given and i ran this via ssh.  ( i am installing on a qnap nas device via the container station)   i have default.opvn file transferred over to the nas too ( i am using airvpn which is a single file) , but i don;t know what directory this needs to go into... as per your question here is the log


"docker pull haugene/transmission-openvpn
 
 docker create --cap-add=NET_ADMIN \
   --name=transmission \
   -v /share/CACHEDEV1_DATA/Downloads:/data \
   -v /etc/localtime:/etc/localtime:ro \
   -v /docker/transmission/watch:/watch \
   -v /share/CACHEDEV1_DATA/docker/transmission/config:/config \
   -e PUID=1000 \
   -e PGID=1000 \
   -e CREATE_TUN_DEVICE=true \
   -e OPENVPN_PROVIDER=asd \
   -e OPENVPN_USERNAME=asd@gmail.com \
   -e OPENVPN_PASSWORD=asdasd \
   -e WEBPROXY_ENABLED=false \
   -e LOCAL_NETWORK=10.0.0.0/24 \
   -e TRANSMISSION_SPEED_LIMIT_DOWN=20000 \
   -e TRANSMISSION_SPEED_LIMIT_DOWN_ENABLED=true \
   -e TRANSMISSION_SPEED_LIMIT_UP=10000 \
   -e TRANSMISSION_SPEED_LIMIT_UP_ENABLED=true \
   -e TRANSMISSION_ALT_SPEED_UP=30000 \
   -e TRANSMISSION_ALT_SPEED_DOWN=60000 \
   -e TRANSMISSION_ALT_SPEED_TIME_ENABLED=true \
   -e TRANSMISSION_ALT_SPEED_TIME_BEGIN=180 \
   -e TRANSMISSION_ALT_SPEED_TIME_END=1020 \
   -e TRANSMISSION_ALT_SPEED_TIME_DAY=62 \
   -e TRANSMISSION_RATIO_LIMIT=1 \
   -e TRANSMISSION_RATIO_LIMIT_ENABLED=true \
   -e TRANSMISSION_DOWNLOAD_QUEUE_SIZE=10 \
   --log-driver json-file \
   --log-opt max-size=10m \
   -p 9091:9091 \
   --restart=always \
   haugene/transmission-openvpn
 
 docker start transmission
 docker logs -f transmission"
